### ISSUE TITLE
Duplicate block issue

### ISSUE THREAD

**Issue by htmgarcia** [publishpress/PublishPress-Blocks#537]
### Steps
1. Create a Countdown block
2. Duplicate

### Result
Both blocks have the the same id, which breaks the duplicated one

### Expected result
Each bock should have different ids

@contactashish13  I think this `if()` is causing the issue, however not sure if is there for a specific reason besides checking if id is `undefined`:
```
// create a blockID and a time of the next day only for a newly inserted block
if(typeof blockID === 'undefined'){
    setAttributes( { blockID: `advgb-countdown-${clientId}`, dateTime: moment().add(1, 'days').format('YYYY-MM-DDTHH:mm:ss') } );
}
```

**Comment by htmgarcia** [comment]
@contactashish13 I think this should do the trick: https://github.com/publishpress/PublishPress-Blocks-Pro/commit/64068701fd75f850d55e89d4644ad2ad5ba1f37b

**Comment by contactashish13** [comment]
@htmgarcia that still duplicates the block with the same ID and it causes this weird behavior that you saw before. The changes I was trying also result in this.


https://user-images.githubusercontent.com/12953439/119770034-40d7d780-bed9-11eb-83de-b7f691468aeb.mp4

**Comment by htmgarcia** [comment]
@contactashish13 following the exact same steps, I get the same result even after reverting my changes.

**Comment by contactashish13** [comment]
@htmgarcia looks like this behavior happens only when we duplicate blocks without saving. If I create a block, save and then duplicate the block it all goes well.

**Comment by htmgarcia** [comment]
@contactashish13 thanks! Do you have any suggestion to fix this?
> looks like this behavior happens only when we duplicate blocks without saving.

**Comment by contactashish13** [comment]
@htmgarcia I've tried to set the new block ID in the state instead of the attributes and use it from there - that seems to work to change the block ID when duplicating but this blinking thing still doesn't go away. I'm trying to fix that. Let me know if you have any suggestions.

**Comment by htmgarcia** [comment]
@contactashish13 any progress on this one?

Regards

**Comment by contactashish13** [comment]
@htmgarcia still trying a variety of things to fix the blinking issue. Will try to figure this out in this week before I give up :)

**Comment by contactashish13** [comment]
@htmgarcia just wanted to let you know that I couldn't take a look at this last week so I will be taking a stab at it today/tomorrow.

**Comment by htmgarcia** [comment]
Thanks @contactashish13 !

**Comment by contactashish13** [comment]
@htmgarcia Finally I have had to give up on this. I have pushed in my changes for you to review: https://github.com/contactashish13/PublishPress-Blocks-Pro/pull/1

When you duplicate a block it gets a new ID but no matter when you duplicate the block (duplicate a newly created block or duplicate a saved block), the timer seems to blink. 

The problem seems to be with the interval not being cleared for some reason in the `clearTimer` function. I have created a singleton pattern because I suspected a duplication was causing multiple instances of the interval object `advgCountdownIntervals` to be created but that doesn't seem to be helping.

Let me know if you are able to crack this. A fresh pair of eyes would definitely help :)
